packages/grafana-runtime/src/accesscontrol/authorize.ts

```diff
@@ -0,0 +1,104 @@
+import { locationService, type BackendSrv } from '../services';
+
+type PermissionMap = Record<string, boolean>;
+
+export interface Authorizer {
+  getSignedInUserPermissions(reload: boolean): Promise<PermissionMap>;
+  hasPermission(principal: AuthorizationPrincipal, action: string): boolean;
+  hasAnyPermission(principal: AuthorizationPrincipal, actions: string[]): boolean;
+  hasAllPermissions(principal: AuthorizationPrincipal, actions: string[]): boolean;
+}
+
+interface AuthorizationPrincipal {
+  permissions?: PermissionMap;
+}
+
+function hasPermission(principal: AuthorizationPrincipal, action: string): boolean {
+  return !!principal.permissions?.[action];
+}
+
+/**
+ * DefaultAuthorizer is one that makes authorization decisions based on passed information.
+ * It would be up to the user of DefaultAuthorizer to decide how to store and provide authorization
+ * information to checks.
+ */
+export class DefaultAuthorizer {
+  private backend: () => BackendSrv;
+
+  constructor(backend: () => BackendSrv) {
+    this.backend = backend;
```

Comment:
Do we really need to pass backend as a param? Couldn't we just call `getBackendSrv()` here?

Reply:
Yeah I wasn't sure what to do. I think depending on `getBackendSrv` is ok, but if you wanted to test this depending on the interface is easier that jest mocking imho.

Reply:
yes, for testing it's seems easier to inject it later